core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+    update_cna_metafile(meta_cna_file_path, pd_file_name)
+    print(" DONE")
+
+    logger.info('Import complete')
+
+    return exit_status_handler.get_exit_status()
+
+
+def open_cna_file(file_name):
+    """Open CNA file and handle exception when not found."""
+    try:
+        file = open(file_name)
+    except FileNotFoundError:
+        raise FilenotFoundError("Could not open discrete CNA file at path '" + file_name + "'")
+    return file
+
+
+def get_features(cna_file_path):
+    """Extract CNA events from CNA data file."""
+    header_elements = libImportOncokb.get_first_line(open_cna_file(cna_file_path)).rstrip('\n').split('\t')
+    header_indexes = {}
+    for required_column in required_cna_columns + ['Entrez_Gene_Id']:
+        header_indexes[required_column] = header_elements.index(required_column)
+    sample_ids = [i for j, i in enumerate(header_elements) if j not in header_indexes.values()]
+    sample_indexes = {}
+    for sample_id in sample_ids:
+        sample_indexes[sample_id] = header_elements.index(sample_id)
+
+    features = []
+    cna_file = open_cna_file(cna_file_path)
+    print("Reading features from file ...", end = '')
+    for line in cna_file:
+        if line == '\n' or line.startswith('#') or line.startswith(header_elements[0]):
+            continue  # skip comment and header line
+        line_elements = line.rstrip('\n').split('\t')
+        for sample_id in sample_ids:
+            feature = {}
+            feature['sample_id'] = sample_id
+            feature['alteration'] = int(line_elements[sample_indexes[sample_id]])
+            # cna value 0 (no CNA) is skipped
+            if feature['alteration'] == 0:
+                continue
+            feature['copyNameAlterationType'] = list(cna_alteration_types.keys())[
+                list(cna_alteration_types.values()).index(feature['alteration'])]
+            for column_name, index in header_indexes.items():
+                value = line_elements[index]
+                if value != '':
+                    feature[column_name] = value
+                elif column_name != 'Entrez_Gene_Id':
+                    print(Color.RED + "Empty value encounterd in column '" + column_name + "' in row " + str(
+                        row_counter) + ". OncoKb annotations cannot be imported. Please fix and rerun." + Color.END,
+                          file=sys.stderr)
+                    print("!" * 71, file=sys.stderr)
+                    sys.exit(1)
+
+            # resolve gene symbols to Entrez Ids if needed
+            if 'Entrez_Gene_Id' in feature and feature['Entrez_Gene_Id'] is not None and feature[
+                'Entrez_Gene_Id'] != '':
+                entrez_gene_ids = [feature['Entrez_Gene_Id']]
+            else:
+                entrez_gene_ids = portal_instance.hugo_entrez_map[feature['Hugo_Symbol']]
+
+            if len(entrez_gene_ids) > 1:
+                logger.error("Multiple Entrez gene ids were found for a gene." \
```

Comment:
I noticed that in some of the one line string, you use three quotes even there is not need to escape the quote. Maybe that's the convention you like to use? Here would be a perfect place to use three quotes with multiple lines?

Reply:
Ah I did not know that `"""` is used for multiline comments :) I have updated where possible.